src/Stream.ts

```diff
@@ -61,6 +61,48 @@ export interface Emitter<A> {
   end(): void
 }
 
+export const fromAsyncIterable: {
+  <A, R>(iterable: AsyncGenerator<A, R>): Fx.Fx<Async.Async | Stream<A>, R>
+  <A>(iterable: AsyncIterable<A>): Fx.Fx<Async.Async | Stream<A>, unknown>
+} = <A>(iterable: AsyncIterable<A>): Fx.Fx<Async.Async | Stream<A>, unknown> => Fx.bracket(
+  Fx.sync(() => iterable[Symbol.asyncIterator]()),
+  iterator => Async.run(() => (iterator.return?.() ?? Promise.resolve()).then(() => { })),
+  iterator => Fx.fx(function* () {
+    let next = yield* Async.run(() => iterator.next())
+    while (!next.done) {
+      yield* event(next.value)
+      next = yield* Async.run(() => iterator.next())
+    }
+    return next.value
```

Comment:
Can probably wrap up the Async once:
```suggestion
    const next = Async.run(() => iterator.next())
    let ir = yield* next
    while (!next.done) {
      yield* event(next.value)
      ir = yield* next
    }
    return ir.value
```

Reply:
Updated https://github.com/briancavalier/fx/pull/5/commits/8d5bac8640084704a004884cbe382b46112df6e7